tools/legacyhguardcheck/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"slices"
+
+	"github.com/cilium/cilium/pkg/safeio"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "Usage: legacyhguardcheck <path>")
+		os.Exit(1)
+	}
+
+	found, err := checkDir(os.Args[1])
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	if found {
+		fmt.Fprintln(os.Stderr, "Found legacy header guards, please replace with #pragma once or "+
```

Comment:
What does the output look like? Does it include line and file number?

Reply:
Good point. I added a line number, it now looks like: `Found legacy header guard with __EXAMPLE_H__ macro at bpf/some_example_file.h:123`